drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1184 @@
+/*
+ * Copyright (c) 2023 deveritec GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_tmag5273
+
+#include "tmag5273.h"
+
+#include <stdlib.h>
+
+#include <zephyr/drivers/sensor/tmag5273.h>
+
+#include <zephyr/init.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/check.h>
+#include <zephyr/sys/crc.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(TMAG5273, CONFIG_SENSOR_LOG_LEVEL);
+
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+	#warning "TMAG5273 driver enabled without any devices"
+#endif
+
+#define TMAG5273_XYZ_MEAS_RANGE_LOW \
+	(FIELD_PREP(TMAG5273_X_Y_RANGE_MSK, TMAG5273_X_Y_MEAS_RANGE_LOW) \
+		| FIELD_PREP(TMAG5273_Z_RANGE_MSK, TMAG5273_Z_MEAS_RANGE_LOW))
+#define TMAG5273_XYZ_MEAS_RANGE_HIGH \
+	(FIELD_PREP(TMAG5273_X_Y_RANGE_MSK, TMAG5273_X_Y_MEAS_RANGE_HIGH) \
+		| FIELD_PREP(TMAG5273_Z_RANGE_MSK, TMAG5273_Z_MEAS_RANGE_HIGH))
+
+/**
+ * @brief read a specific \a var from a buffer \a buf and write it to \a out
+ */
+#define SET_RESULT_FROM_BUFFER(buf, var, out) \
+	out = buf[TMAG5273_REG_##var##_MSB_RESULT-TMAG5273_REG_RESULT_BEGIN] \
+		& TMAG5273_##var##_RESULT_MSB_MSK; \
+	out = out << 8; \
+	out |= (buf[TMAG5273_REG_##var##_LSB_RESULT-TMAG5273_REG_RESULT_BEGIN] \
+		& TMAG5273_##var##_RESULT_LSB_MSK);
+
+#define CONV_FACTOR_MT_TO_GS (10)
+
+/**
+ * @brief resets the DEVICE_STATUS register
+ *
+ * @param dev driver handle
+ * @retval see @ref i2c_reg_write_byte
+ */
+static int tmag5273_reset_device_status(const struct device *dev)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+
+	return i2c_reg_write_byte_dt(&drv_cfg->i2c,
+		TMAG5273_REG_DEVICE_STATUS,
+		0xFF);
+}
+
+/**
+ * @brief reads the DEVICE_STATUS register
+ *
+ * Will also print a human readable representation to the log, if activated.
+ *
+ * @param dev driver handle
+ *
+ * @retval register value on success (>= 0)
+ * @retval see @ref i2c_reg_read_byte for error codes
+ */
+static int tmag5273_read_device_status(const struct device *dev)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	uint8_t dev_stat = 0;
+	int retval = i2c_reg_read_byte_dt(&drv_cfg->i2c,
+			TMAG5273_REG_CONV_STATUS,
+			&dev_stat);
+
+	if (retval == 0) {
+		retval = dev_stat;
+
+		if (FIELD_GET(dev_stat, TMAG5273_VCC_UV_ER_MSK) == TMAG5273_VCC_UV_ERR) {
```

Comment:
You can probably simplify this to 
```suggestion
		if (dev_stat & TMAG5273_VCC_UV_ER_MSK) {
```

Reply:
Well, the intention here is to have some code which speaks for itself.
Granted, it would work this way here, but this would create a special case just for this check which more often than not increased the confusion.